packages/itwin/property-grid/src/hooks/UseContextMenu.tsx

```diff
@@ -81,18 +92,24 @@ export interface FavoritePropertiesContextMenuItemProps extends ContextMenuItemP
  * Renders `Add to Favorite` context menu item if property field is not favorite. Otherwise renders nothing.
  * @public
  */
-export function AddFavoritePropertyContextMenuItem({ field, imodel, scope }: FavoritePropertiesContextMenuItemProps) {
+export function AddFavoritePropertyContextMenuItem({ field, imodel, scope, onSelect }: FavoritePropertiesContextMenuItemProps) {
   const currentScope = scope ?? FavoritePropertiesScope.IModel;
   if (!field || Presentation.favoriteProperties.has(field, imodel, currentScope)) {
     return null;
   }
 
+  const defaultAction = async () => Presentation.favoriteProperties.add(field, imodel, currentScope);
+
   return (
     <PropertyGridContextMenuItem
       id="add-favorite"
       onSelect={async () => {
-        assert(field !== undefined);
-        await Presentation.favoriteProperties.add(field, imodel, currentScope);
+        if (onSelect) {
+          onSelect(defaultAction);
```

Comment:
`defaultAction` here is async, but we pass it as non-async to `onSelect`. Maybe `onSelect` should accept an async function to allow consumers await on the `defaultAction` if necessary?